core/tests/test_utils.py

```diff
@@ -861,6 +861,42 @@ def post_json(
         self.assertEqual(json_response.status_int, expected_status_int)
         return self._parse_json_response(json_response, expect_errors)
 
+    def post_blob(self, url, payload, csrf_token=None, expected_status_int=200):
+        """Post a BLOB object to the server; return the received object.
+
+        Args:
+            url: str. The URL to which BLOB object in payload should be sent
+                through a post request.
+            payload: bytes. Binary data which needs to be sent.
+            csrf_token: str. CSRF token.
+            expected_status_int: int. The status expected as a response of post
+                request.
+
+        Returns:
+            dict. Parsed JSON response received upon invoking the post request.
+        """
+        data = payload
+        if csrf_token:
+            data['csrf_token'] = csrf_token
```

Comment:
Can this be done if the payload is a bytes object?

Also I'm a bit confused about why we need this. How have we been handling asset posts before? We should probably use the same methods instead.

Reply:
No, it can't be done. Also in other assets such as Audio and Image, we use JavaScript (FormData) object, to transfer data which basically uses Base64 encoding and transfers it to backend. I suggest we don't use Base64 for receiving post payloads from Oppia-ml because it will increase the size of payload.